modules/sdk-core/src/bitgo/utils/opengpgUtils.ts

```diff
@@ -32,6 +32,33 @@ export async function getBitgoGpgPubKey(bitgo: BitGoBase): Promise<Key> {
   return await readKey({ armoredKey: bitgoPublicKeyStr });
 }
 
+/**
+ * Verify an Eddsa KeyShare Proof.
+ *
+ * @param senderPubKey public key of the sender of the privateShareProof
+ * @param privateShareProof u value proof
+ * @param uValue u value from an Eddsa keyshare
+ * @return {boolean} whether uValue proof actually was signed by sender as part of their subkeys
+ */
+export async function verifyShareProof(
+  senderPubKey: string,
+  privateShareProof: string,
+  uValue: string
+): Promise<boolean> {
+  const decodedProof = await pgp.readKey({ armoredKey: privateShareProof });
+  const otherOvcGpgKey = await pgp.readKey({ armoredKey: senderPubKey });
+  const isValid = (await decodedProof.verifyPrimaryUser([otherOvcGpgKey]))[0].valid;
+  if (!isValid) {
+    return false;
+  }
+  const proofSubkeys = decodedProof.getSubkeys()[1];
+  const decodedUValueProof = Buffer.from(proofSubkeys.keyPacket.publicParams['Q'].slice(1)).toString('hex');
+  const rawUValueProof = Buffer.from(
+    sodium.crypto_scalarmult_ed25519_base_noclamp(Buffer.from(uValue, 'hex'))
```

Comment:
when we deploy ECDSA TSS, we will need the equivalent math operation here.    which means we will probably need to make separate proof functions for ed and ec.    I suggest changing the function name to something like verifyEdShareProof or something along those lines.